operator/api/v1beta1/lokistack_types.go

```diff
@@ -253,6 +253,18 @@ type LokiComponentSpec struct {
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
 }
 
+// LokiRulerComponentSpec defines the requirements to configure ruler component
+type LokiRulerComponentSpec struct {
+	LokiComponentSpec `json:",inline"`
+
+	// PrometheusEndpoint defines the recording rules metrics push endpoint
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:hidden"
+	PrometheusEndpoint string `json:"prometheusEndpoint,omitempty"`
```

Comment:
I can't find where this is used. @jacobsakran can you explain what we are doing with this field? why do we need this?

Reply:
This `spec` should probably be a part of the `LokiStackSpec`. However, what I am unclear about is how many pieces of the [configuration](https://grafana.com/docs/loki/latest/configuration/#ruler_config) need to be exposed to make this properly configurable for a user.

Reply:
I think we talked about that @eranra in the meeting